homeassistant/components/modbus/validators.py

```diff
@@ -388,7 +390,11 @@ def validate_entity(
                     del entities[entity_inx]
                 else:
                     entity_inx += 1
-
+        if no_entities:
+            err = f"Modbus {hub[CONF_NAME]} is instable without entities, please add entities!"
+            _LOGGER.warning(err)
+            # Ensure timeout is not started/handled.
+            hub[CONF_TIMEOUT] = -1
```

Comment:
So for whom would this log message be? Because if someone adds this to support a custom integration, the message to "please add entities!" doesn't really make sense, since the custom integration is doing this for them.

In what way will it be unstable?

Maybe sentences like:

"There are no entities defined for {something}. Ignoring timeout."
"There are no entities defined for {something}. This can lead to modbus updating less predictable."

Reply:
The text is for the end-user who uses the integration, and even if the custom integration adds entities they, we have no idea how (and do not want to know), so we have no way of knowing if the integration continues to be unstable.

The only valid option is to add at least one entity (some custom components are now, finally, starting to recommend that).

I have tried to adjust the text along your suggestion. WDYT ?